community/modules/scripts/ramble-setup/templates/ramble_setup.yml.tftpl

```diff
@@ -54,7 +54,10 @@
     failed_when: false
 
   - name: Clones into installation directory
-    ansible.builtin.command: git clone --branch {{ git_ref }} {{ git_url }} {{ install_dir }}
+    ansible.builtin.command: "{{ item }}"
+    with_items:
+      - git clone {{ git_url }} {{ install_dir }}
+      - git --git-dir={{ install_dir }}/.git checkout {{ git_ref }}
```

Comment:
While this looks more elegant there is a performance cost for the happy path. I honestly know just enough to be dangerous, but performing the checkout operation can be slow, especially when it happens over nfs. When we clone the repo without the `--branch` arg it will checkout the default branch. So basically we now have to do 2x checkouts for a git clone. 

I would go back to the logic of trying to clone the branch and if that doesn't work then clone and checkout the branch.

Open to discussion especially if you have new data.

Reply:
Done